hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/Constants.java

```diff
@@ -1276,4 +1276,16 @@ private Constants() {
   public static final String STORE_CAPABILITY_DIRECTORY_MARKER_MULTIPART_UPLOAD_ENABLED =
       "fs.s3a.capability.multipart.uploads.enabled";
 
+  /**
+   * Prefetch max blocks count config.
+   * Value = {@value PREFETCH_MAX_BLOCKS_COUNT}
+   */
+  public static final String PREFETCH_MAX_BLOCKS_COUNT = "fs.s3a.prefetch.max.blocks.count";
+
+  /**
+   * Default value for max blocks count config.
+   * Value = {@value DEFAULT_PREFETCH_MAX_BLOCKS_COUNT}
+   */
+  public static final int DEFAULT_PREFETCH_MAX_BLOCKS_COUNT = 10;
```

Comment:
this is per stream? it's big. we may want to see what happens in apps with many threads -risk of out of disk is high.

Reply:
perhaps 4 is enough?